### PR TITLE
Add pagination to failures view

### PR DESCRIPTION
**Pull request by jscook2345**
Use the FailuresWithConfidence class to control database access,
pagination, and searching for the failures view.

This class provides these features to improve performance when gathering
failing tests and their confidence score.

### CODE REVIEW

File: squad/frontend/templates/squad/failures.jinja2 [hunk 1]
```diff
@@ -23,26 +27,27 @@
             <th>{{ e.slug }}</th>
             {% endfor %}
         </thead>
-        {% if not ufailures %}
+        {% if fc.failures()|length == 0 %}
```
Comment (chaws) [inline squad/frontend/templates/squad/failures.jinja2:30]: `fc.failures()` is the heaviest query and you are already calling it on line 21 in squad/frontend/failures.py

Comment (jscook2345) [reply squad/frontend/templates/squad/failures.jinja2:30]: I changed this to `fc.failures().exists()` instead. Thanks for catching this.

Comment (chaws) [reply squad/frontend/templates/squad/failures.jinja2:30]: Does `.exists()` re-use the same query? I think it'll still re-run the query + `LIMIT 1`

File: squad/core/failures.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def number(self):
+        return self.page
+
+    def failures(self):
+        page = self.paginator.page(self.page)
+
+        failures = self.build.tests.filter(
+            result=False,
+            suite__slug__in=[t.suite__slug for t in page.object_list],
+            metadata__name__in=[t.metadata__name for t in page.object_list],
```
[the remark this answers is not in the document]
Comment (jscook2345) [reply squad/core/failures.py]: I implemented this as suggested.

**Comment by chaws** [comment]
LGTM

**Pull request merged**